lib/galaxy/schema/schema.py

```diff
@@ -1896,12 +1896,24 @@ class ArchivedHistoryDetailed(HistoryDetailed, ExportAssociationData):
     pass
 
 
-AnyArchivedHistoryView = Union[
-    ArchivedHistorySummary,
-    ArchivedHistoryDetailed,
-    # Any will cover those cases in which only specific `keys` are requested
-    # otherwise the validation will fail because the required fields are not returned
-    Any,
+@partial_model()
+class CustomArchivedHistoryView(CustomHistoryView, ExportAssociationData):
+    """Archived History Response with all optional fields.
+
+    It is used for serializing only specific attributes using the "keys"
+    query parameter.
+    """
+
+    pass
+
+
+AnyArchivedHistoryView = Annotated[
+    Union[
+        CustomArchivedHistoryView,
+        ArchivedHistoryDetailed,
+        ArchivedHistorySummary,
+    ],
+    Field(union_mode="left_to_right"),
```

Comment:
Why do we validate the Custome one first with left to right ? Shouldn't it be Detailed -> Summary -> Custom ?

Reply:
I basically replicated what we used for `AnyHistoryView` in https://github.com/galaxyproject/galaxy/pull/17779

I remember struggling exactly with the same question and changing this order several times until everything "worked", but let me do some additional tests to confirm.

Reply:
I see, so we are in fact always returning custom models at runtime and we're only using the other models for the documentation?